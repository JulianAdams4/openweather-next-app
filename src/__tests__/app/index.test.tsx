import '@root/__mocks__/matchMedia.mock';
import { render } from '@testing-library/react';

import MainPage from '@app/page';

const unmockedFetch = global.fetch;

beforeAll(() => {
  global.fetch = jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve({ test: 100 }),
    })
  ) as jest.Mock;
});

afterAll(() => {
  global.fetch = unmockedFetch;
});

it('Renders main page with loader', () => {
  const { container } = render(<MainPage />);
  expect(container).toMatchSnapshot();
});
