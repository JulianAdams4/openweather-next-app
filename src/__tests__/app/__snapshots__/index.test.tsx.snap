// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders main page with loader 1`] = `
<div>
  <div
    style="flex: 1; position: absolute; top: 0px; left: 0px; display: flex; justify-content: center; align-items: center; background-image: url(/img.jpg); background-size: cover; width: 100%; height: 100%; z-index: 10;"
  >
    <p
      style="font-size: 2.8rem;"
    >
      Cargando...
    </p>
  </div>
  <div
    class="ant-layout css-dev-only-do-not-override-djtmh8"
    style="width: 100%; min-height: 100vh;"
  >
    <div
      class="ant-row css-dev-only-do-not-override-djtmh8"
      style="flex: 1;"
    >
      <div
        class="ant-col ant-col-xs-0 ant-col-sm-10 ant-col-md-8 ant-col-lg-6 ant-col-xl-5 css-dev-only-do-not-override-djtmh8"
      >
        <div
          class="ant-col css-dev-only-do-not-override-djtmh8"
          style="height: 100%; justify-content: space-between; background-color: rgb(91, 109, 125); overflow-y: scroll; padding: .5em;"
        >
          <h3
            class="ant-typography css-dev-only-do-not-override-djtmh8"
            style="color: rgb(255, 255, 255); font-weight: normal; padding: .6rem .7em;"
          >
            Favoritos
          </h3>
          <div
            class="ant-col css-dev-only-do-not-override-djtmh8"
            style="flex: 1 1 auto; padding: 1.5em .2em;"
          >
            <div
              class="ant-row css-dev-only-do-not-override-djtmh8"
            >
              <span
                class="ant-typography css-dev-only-do-not-override-djtmh8"
                style="font-size: 1.1rem; color: rgb(255, 255, 255); padding: .5rem;"
              >
                Aún no has agreagado Favoritos.
              </span>
            </div>
            <div
              class="ant-row css-dev-only-do-not-override-djtmh8"
            >
              <span
                class="ant-typography css-dev-only-do-not-override-djtmh8"
                style="font-size: 1.1rem; color: rgb(255, 255, 255); padding: .5rem;"
              >
                Intenta presionando el botón  
                <span
                  aria-label="star"
                  class="anticon anticon-star"
                  role="img"
                  style="font-size: 16px; color: rgb(247, 186, 3);"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="star"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                    />
                  </svg>
                </span>
                 junto al nombre de una ciudad
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-xs-24 ant-col-sm-14 ant-col-md-16 ant-col-lg-18 ant-col-xl-19 css-dev-only-do-not-override-djtmh8"
        style="background-image: url(/img.jpg); background-size: cover;"
      >
        <div
          class="ant-row css-dev-only-do-not-override-djtmh8"
          style="height: 64px; display: flex; align-items: center; justify-content: space-between; background-color: transparent; padding-inline: 1em;"
        >
          <div
            class="ant-col ant-col-xs-2 ant-col-sm-0 css-dev-only-do-not-override-djtmh8"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-djtmh8 ant-btn-link ant-btn-lg ant-btn-icon-only"
              disabled=""
              style="color: rgb(255, 255, 255);"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <span
                  aria-label="menu"
                  class="anticon anticon-menu"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="menu"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
          <div
            class="ant-col ant-col-xs-22 ant-col-sm-24 css-dev-only-do-not-override-djtmh8"
            style="display: flex; flex-direction: row-reverse;"
          >
            <div
              class="ant-col ant-col-xs-19 ant-col-sm-16 ant-col-md-11 ant-col-lg-8 ant-col-xl-5 css-dev-only-do-not-override-djtmh8"
            >
              <span
                class="ant-input-affix-wrapper ant-input-disabled ant-input-affix-wrapper-disabled ant-input-affix-wrapper-lg css-dev-only-do-not-override-djtmh8 ant-input-borderless"
                style="color: rgb(255, 255, 255); background-color: rgba(0, 0, 0, 0.2);"
              >
                <span
                  class="ant-input-prefix"
                >
                  <span
                    aria-label="search"
                    class="anticon anticon-search"
                    role="img"
                    style="margin-right: 10px; opacity: 0.5;"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="search"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                      />
                    </svg>
                  </span>
                </span>
                <input
                  class="ant-input ant-input-disabled ant-input-lg search-input-input css-dev-only-do-not-override-djtmh8"
                  disabled=""
                  maxlength="30"
                  placeholder="Ingresa una ciudad"
                  type="text"
                  value=""
                />
                <span
                  class="ant-input-suffix"
                >
                  <span
                    class="ant-input-clear-icon ant-input-clear-icon-hidden ant-input-clear-icon-has-suffix"
                    role="button"
                    tabindex="-1"
                  >
                    <span
                      aria-label="close-circle"
                      class="anticon anticon-close-circle"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="close-circle"
                        fill="currentColor"
                        fill-rule="evenodd"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="ant-input-show-count-suffix search-input-count"
                  >
                    0 / 30
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-row-center css-dev-only-do-not-override-djtmh8"
          style="flex: 1; height: 100%; max-height: calc(100% - 64px);"
        />
      </div>
    </div>
  </div>
</div>
`;
